### ISSUE TITLE
SublimeLinter regression following PR #637

### ISSUE THREAD

**Issue by jrappen** [SublimeLinter/SublimeLinter3#641]
After starting SublimeLInter I get this error:

```text
SublimeLinter: error in SublimeLinter daemon: 
SublimeLinter: -------------------- 
SublimeLinter: Traceback (most recent call last):
  File "...\SublimeLinter\lint\queue.py", line 57, in loop
    item = self.q.get(block=True, timeout=self.MIN_DELAY)
  File "./python3.3/queue.py", line 175, in get
queue.Empty

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "...\SublimeLinter\lint\queue.py", line 65, in loop
    self.lint(view_id, timestamp)
  File "...\SublimeLinter\lint\queue.py", line 111, in lint
    self.callback(view_id, timestamp)
  File "...\SublimeLinter\sublimelinter.py", line 120, in lint
    Linter.lint_view(view, filename, code, hit_time, callback)
  File "...\SublimeLinter\lint\linter.py", line 924, in lint_view
    view_settings = linter.get_view_settings(inline=False)
  File "./python3.3/functools.py", line 251, in wrapper
  File "...\SublimeLinter\lint\linter.py", line 449, in get_view_settings
    settings = self.get_merged_settings()
  File "...\SublimeLinter\lint\linter.py", line 496, in get_merged_settings
    self.replace_settings_tokens(settings)
  File "...\SublimeLinter\lint\linter.py", line 594, in replace_settings_tokens
    if folder['path'] in view.file_name():
TypeError: argument of type 'NoneType' is not iterable
```

which comes from changes introduced in PR #637.

https://github.com/SublimeLinter/SublimeLinter3/pull/637/files#diff-8120b617e4c6c91f43e1aec1ba391da7R593

https://github.com/SublimeLinter/SublimeLinter3/blob/1319d102b383b64aa9642cd8edb67fcf46e39847/lint/linter.py#L594

**Comment by groteworld** [comment]
Can you give us an example or steps of how to replicate this issue? Do you have no file open, an unsaved file open, a single file open, or a project open?

**Comment by sebastienbarre** [comment]
Thanks for filing a bug, my SublimeLinter has been broken too since a recent update, couldn't pinpoint which, really trying to revert back to this release, if I find a way. My error message is slightly different:
```
environment variables loaded using: /bin/bash -l
SublimeLinter: error in SublimeLinter daemon: 
SublimeLinter: -------------------- 
SublimeLinter: Traceback (most recent call last):
  File "/Users/barre/Library/Application Support/Sublime Text 3/Packages/SublimeLinter/lint/queue.py", line 57, in loop
    item = self.q.get(block=True, timeout=self.MIN_DELAY)
  File "./python3.3/queue.py", line 175, in get
queue.Empty

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/barre/Library/Application Support/Sublime Text 3/Packages/SublimeLinter/lint/queue.py", line 65, in loop
    self.lint(view_id, timestamp)
  File "/Users/barre/Library/Application Support/Sublime Text 3/Packages/SublimeLinter/lint/queue.py", line 111, in lint
    self.callback(view_id, timestamp)
  File "/Users/barre/Library/Application Support/Sublime Text 3/Packages/SublimeLinter/sublimelinter.py", line 120, in lint
    Linter.lint_view(view, filename, code, hit_time, callback)
  File "/Users/barre/Library/Application Support/Sublime Text 3/Packages/SublimeLinter/lint/linter.py", line 924, in lint_view
    view_settings = linter.get_view_settings(inline=False)
  File "./python3.3/functools.py", line 251, in wrapper
  File "/Users/barre/Library/Application Support/Sublime Text 3/Packages/SublimeLinter/lint/linter.py", line 449, in get_view_settings
    settings = self.get_merged_settings()
  File "/Users/barre/Library/Application Support/Sublime Text 3/Packages/SublimeLinter/lint/linter.py", line 496, in get_merged_settings
    self.replace_settings_tokens(settings)
  File "/Users/barre/Library/Application Support/Sublime Text 3/Packages/SublimeLinter/lint/linter.py", line 594, in replace_settings_tokens
    if folder['path'] in view.file_name():
KeyError: 'path'
 ```

**Comment by groteworld** [comment]
@sebastienbarre your issue seems related to this issue, but is slightly different, noticing your issue is that the 'path' key is incorrect. Could you provide some context as well to how I can replicate this issue?

**Comment by sebastienbarre** [comment]
Yes I noticed the end was slightly different. I can try to give you as much info as possible, but it triggers as soon as I open a file that would other be linting fine not so long ago. And once triggered, it just ceases to lint.

I've tried to uninstall/reinstall the package, no change.

Here are my settings:
```yaml
{
    "user": {
        "debug": false,
        "delay": 2,
        "error_color": "0000a0",
        "gutter_theme": "Packages/SublimeLinter/gutter-themes/Default/Default.gutter-theme",
        "gutter_theme_excludes": [],
        "lint_mode": "load/save",
        "linters": {
            "csslint": {
                "@disable": false,
                "args": [],
                "errors": "",
                "excludes": [
                    "*.ctp"
                ],
                "ignore": "",
                "warnings": ""
            },
            "eslint": {
                "@disable": false,
                "args": [],
                "excludes": [
                    "*.html"
                ]
            },
            "flow": {
                "@disable": false,
                "args": [],
                "excludes": [],
                "flow-bin": null,
                "show-all-errors": true
            },
            "htmlhint": {
                "@disable": false,
                "args": [],
                "excludes": [
                    "*.ctp",
                    "*.php"
                ]
            },
            "php": {
                "@disable": false,
                "args": [],
                "excludes": [
                    "*.html"
                ]
            },
            "phpcs": {
                "@disable": false,
                "args": [],
                "excludes": [
                    "*.html"
                ],
                "standard": ""
            },
            "phpmd": {
                "@disable": true,
                "args": [],
                "excludes": [
                    "*.html"
                ],
                "rulesets": "${project}/phpmd.xml"
            }
        },
        "mark_style": "fill",
        "no_column_highlights_line": true,
        "passive_warnings": false,
        "paths": {
            "linux": [],
            "osx": [],
            "windows": []
        },
        "python_paths": {
            "linux": [],
            "osx": [],
            "windows": []
        },
        "rc_search_limit": 3,
        "shell_timeout": 10,
        "show_errors_on_save": false,
        "show_marks_in_minimap": true,
        "syntax_map": {
            "coffeescript (gulpfile)": "coffeescript",
            "ctp": "html",
            "html (django)": "html",
            "html (rails)": "html",
            "html 5": "html",
            "javascript (babel)": "javascript",
            "javascript (eslint)": "javascript",
            "javascript (gruntfile)": "javascript",
            "javascript (gulpfile)": "javascript",
            "javascript (postcss)": "javascript",
            "javascript (stylelint)": "javascript",
            "javascript (webpack)": "javascript",
            "json (babel)": "json",
            "json (bower)": "json",
            "json (composer)": "json",
            "json (eslint)": "json",
            "json (npm)": "json",
            "json (postcss)": "json",
            "json (settings)": "json",
            "json (stylelint)": "json",
            "json (sublime)": "json",
            "json (tern js)": "json",
            "php": "html",
            "python django": "python",
            "xml (config)": "xml",
            "xml (svg)": "xml",
            "yaml (circleci)": "yaml",
            "yaml (docker)": "yaml",
            "yaml (eslint)": "yaml",
            "yaml (lock)": "yaml",
            "yaml (procfile)": "yaml",
            "yaml (stylelint)": "yaml",
            "yaml (yarn)": "yaml"
        },
        "tooltip_fontsize": "0.9rem",
        "tooltip_theme": "Packages/SublimeLinter/tooltip-themes/Default/Default.tooltip-theme",
        "tooltip_theme_excludes": [],
        "tooltips": false,
        "use-server": false,
        "warning_color": "DDB700",
        "wrap_find": true
    }
}
```

**Comment by jrappen** [comment]
* I opened Sublime
* Opened a new tab
* Pasted some code
* Syntax auto-detected, SublimeLinter started, Linter loaded
* Error above

...so it was a single unsaved file.

**Comment by kiddick** [comment]
same here

**Comment by asfaltboy** [comment]
I also met with a similar issue, but with a different side-effect; the flake8 linter stopped working with a weird error:

SublimeLinter: flake8 linter reloaded 
reloading plugin SublimeLinter_flake8.linter
SublimeLinter: WARNING: no flake8 version could be extracted from:
Traceback (most recent call last):
  File "/Users/pavel/.pyenv/versions/2.7.11/bin/flake8", line 7, in <module>
    from flake8.main.cli import main
ImportError: No module named flake8.main.cli

I resolved it by adding my locations of flake8 to paths in `SublimeLinter.sublime-settings - User`:
```json
"paths": {
    "linux": [],
    "osx": [
        "/Users/pavel/.pyenv/versions/2.7.11/bin/",
        "/Users/pavel/.pyenv/versions/3.5.2/bin/"
    ],
    "windows": []
}
```

**Comment by groteworld** [comment]
v3.10.2 should fix this issue, Please let me know ASAP if you are still experience this error

**Comment by kylebebak** [comment]
@groteworld
I just upgraded to SublimeLinter 3.10.3, and SublimeLinter-flake8 is still broken in the same way described by @asfaltboy .

I don't know whether this commit was meant to fix that issue, but it didn't. Perhaps this is an issue with SublimeLinter-flake8.

```
reloading plugin SublimeLinter-flake8.linter
SublimeLinter: flake8 version query: /usr/local/bin/python /usr/local/bin/flake8 --version 
SublimeLinter: WARNING: no flake8 version could be extracted from:
Traceback (most recent call last):
  File "/usr/local/bin/flake8", line 7, in <module>
    from flake8.main.cli import main
ImportError: No module named flake8.main.cli
 
SublimeLinter:           using cmd: ['/usr/local/bin/python', '/usr/local/bin/flake8', '--version'], env: {'HOME': '/Users/kylebebak', 'PATH': '/usr/local/bin/python3:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/go/bin', 'USER': 'kylebebak', 'XPC_FLAGS': '0x0', 'LOGNAME': 'kylebebak', 'SSH_AUTH_SOCK': '/private/tmp/com.apple.launchd.jgK4taO1NP/Listeners', 'TMPDIR': '/var/folders/qn/vf83zn0j71vfp27nnd8mn1r80000gn/T/', 'PYTHONIOENCODING': 'utf8', 'XPC_SERVICE_NAME': 'com.sublimetext.3.9828', 'LANG': 'en_US.UTF-8', 'LC_CTYPE': 'en_US.UTF-8', 'Apple_PubSub_Socket_Render': '/private/tmp/com.apple.launchd.pGX3IzBwsz/Render', '__CF_USER_TEXT_ENCODING': '0x1F5:0x0:0x0', 'SHELL': '/bin/zsh'} 
SublimeLinter: flake8 linter reloaded 
```

**Comment by asfaltboy** [comment]
@kylebebak I think this is not really related, your issue is more akin to SublimeLinter/SublimeLinter-flake8#56, probably caused by merge of SublimeLinter/SublimeLinter-flake8#51.

Please ensure you have flake8 installed in python env of `/usr/local/bin/python` or otherwise configure explicit paths / PATH / etc (as in docs).

**Comment by kylebebak** [comment]
@asfaltboy @groteworld 
I tried various suggestions in <https://github.com/SublimeLinter/SublimeLinter-flake8/issues/56> and none worked, except for **joeycarter**'s suggestion to modify the flake8 plugin. A number of people in the same thread reached the same conclusion.

I disagree with the implication that this issue can be resolved by looking at the docs.

Also, installing `flake8` with the `pip` bundled with `/usr/local/bin/python` isn't a good solution, because that's Python2 on macOS.

I don't want to hijack this issue, [so I opened a new one](https://github.com/SublimeLinter/SublimeLinter3/issues/653) to suggest another way of fixing this problem, and other problems like it.

**Comment by sebastienbarre** [comment]
This fixed it for me, thank you. I didn't realize how much I rely on this package, is there a way to lock the version, or more easily revert to a working version, should breakage happen again?

**Comment by groteworld** [comment]
You can always uninstall and just clone the repo into your SublimeText/Packages directory, which will keep it at a specific commit. then you can choose to pull latest when you want to upgrade. but you may miss out on great new features as they come out if you aren't watching. Sorry that this bug happened!

**Comment by jrappen** [comment]
@groteworld This was fixed for me in `v3.10.2` as well, closing issue.
